index.js

```diff
@@ -223,7 +244,8 @@ DB.prototype.batch = function (rows, opts, cb) {
         batch.push(row)
         if (--pending === 0) done()
       } else if (row.type === 'del') {
-        self._del(key, xtend(opts, row), function (err, xrows) {
+        var xrow = xtend(opts, row)
+        self._getDocumentDeletionBatchOps(key, xrow, function (err, xrows) {
```

Comment:
Hah, oh geez, how did this happen? I'll fix this by having `del()` just send a batch row to `batch` so all deletion lookups can happen in one spot.